### PR TITLE
expand oci-image-verification example for private infra

### PR DESCRIPTION
**Pull request by dmitris**
#### Summary
PR expands the [oci-image-verification](https://github.com/sigstore/sigstore-go/blob/main/examples/oci-image-verification/main.go) example with additional options for running it in a "private infrastructure" environment and to allow to extract non-Rekor-based timestamps from the OCI image being verified.

This is done in the mode of exploring the sigstore-go library and for now is a Draft - but I would appreciate any comments especially on whether the timestamp extraction is done correctly.  When trying the `oci-image-verification` on an internally signed image (with TSA), I'm getting an error going back to ASN.1 parsing error (see my question on [Slack](https://sigstore.slack.com/archives/C0440BFT43H/p1729613321322739)):
```bash
$ oci-image-verification -onlineTlog=false -requireTlog=false -requireTimestamp=true -expectedIssuerRegex='.*' -trustedrootJSONpath=$HOME/dev/files/trustedroot.json -expectedSAN='spiffe://cd.internal.project/xyz/12345' -ociImage docker.acompany.com:4443/name/el8/abc:8.10
failed to verify timestamps: threshold not met for verified signed & log entry integrated timestamps: 0 < 1
```
as mentioned in the Slack message, the "underlying" error is (???)
```
ASN.1 parsing error  from [github.com/digitorus/timestamp.ParseResponse](http://github.com/digitorus/timestamp.ParseResponse) (called from [github.com/sigstore/timestamp-authority/pkg/verification.VerifyTimestampResponse](http://github.com/sigstore/timestamp-authority/pkg/verification.VerifyTimestampResponse)):
error parsing response into Timestamp: asn1: structure error: tags don't match (16 vs {class:1 tag:13 length:73 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} response @2
```
#### Release Note
NONE

#### Documentation
n/a

### CODE REVIEW

File: examples/oci-image-verification/main.go [hunk 1]
```diff
@@ -408,6 +423,29 @@ func getVerificationMaterialTlogEntries(manifestLayer *v1.Descriptor) ([]*protor
...
+	}
+	// 3. Verify the key "SignedRFC3161Timestamp" is present
+	if _, ok := keyValPairs["SignedRFC3161Timestamp"]; !ok {
+		return nil, errors.New("error getting SignedRFC3161Timestamp from key/value pairs")
+	}
+	// 4. Construct the timestamp entry list
+	return &protobundle.TimestampVerificationData{
+		Rfc3161Timestamps: []*protocommon.RFC3161SignedTimestamp{
+			{
+				SignedTimestamp: []byte(keyValPairs["SignedRFC3161Timestamp"]),
```
Comment (kommendorkapten) [inline examples/oci-image-verification/main.go]: I think you would need to do `base64.StdEncoding.DecodeString(keyValPairs["SignedRFC3161Timestamp"]` here.

Comment (kommendorkapten) [reply examples/oci-image-verification/main.go]: The json data in the manifest should be base64 decoded, but the protobundle struct expects the raw bytes.

Comment (dmitris) [reply examples/oci-image-verification/main.go]: changed - and it seemed to help, thanks! 😄 , no longer having the asn.1 parsing error (tracing a different one instead - `failed to verify leaf certificate: leaf certificate verification failed`)

**Comment by dmitris** [comment]
was able to verify an OCI image using trustedroot.json, the command was similar to this one, therefore changing from Draft to "Ready to review":
```bash
$ oci-image-verification -ociImage docker.internal.com:4443/name/xyz:1.0 -onlineTlog=false -requireTlog=false -ignore-sct=true -requireTimestamp=true -expectedIssuerRegex='.*' -expectedSAN='spiffe://cd.cisystem.project/cd/build12345' -trustedrootJSONpath=trustedroot.json
Verification successful!
{
   "mediaType": "application/vnd.dev.sigstore.verificationresult+json;version=0.1",
   "signature": {
      "certificate": {
         "certificateIssuer": "CN=[common name]",
         "subjectAlternativeName": "spiffe://cd.cisystem.project/cd/build12345"
      }
   },
   "verifiedTimestamps": [
      {
         "type": "TimestampAuthority",
         "uri": "",
         "timestamp": "2024-10-10T01:28:22Z"
      }
   ],
   "verifiedIdentity": {
      "subjectAlternativeName": {
         "subjectAlternativeName": "spiffe://cd.cisystem.project/cd/build12345"
      },
      "issuer": {
         "issuer": "",
         "regexp": ".*"
      }
   }
}
```

**Review by kommendorkapten — APPROVED** [review]
This looks great, thanks!

**Pull request merged**